server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -183,4 +183,11 @@ public final String simpleName() {
      * so that the current mapper is not modified.
      */
     public abstract Mapper updateFieldType(Map<String, MappedFieldType> fullNameToFieldType);
+
+    /**
+     * Called just after a {@link RootObjectMapper} receives all contained mappers.
+     * @param root The enclosing {@link RootObjectMapper}
+     */
+    protected void rootObjectMapper(final RootObjectMapper root) {
+    }
```

Comment:
This callback happens after all fields are processed and known to the context, and only then can we ask the alias to do its checks, does the target field exist etc.